lib/core/text/flat.nit

```diff
@@ -1308,37 +1308,44 @@ private class FlatBufferCharIterator
 end
 
 redef class CString
-	redef fun to_s
-	do
-		return to_s_with_length(cstring_length)
-	end
 
-	redef fun to_s_with_length(length)
-	do
-		assert length >= 0
-		return clean_utf8(length)
-	end
+	# Get a `String` from the data at `self` copied into Nit memory
+	#
+	# Require: `self` is a null-terminated string.
+	redef fun to_s do return to_s_unsafe
 
-	redef fun to_s_full(byte_length, unilen) do
-		return new FlatString.full(self, byte_length, 0, unilen)
-	end
+	# Get a `String` from `byte_length` bytes at `self` copied into Nit memory
+	#
+	# The string is cleaned.
+	fun to_s_with_length(byte_length: Int): String do return to_s_unsafe(byte_length)
 
-	redef fun to_s_unsafe(len) do
-		if len == null then len = cstring_length
-		return new FlatString.with_infos(self, len, 0)
-	end
+	redef fun to_s_unsafe(byte_length, char_length, copy, clean)
```

Comment:
My bad, I thought the `unsafe` here would mean that it was completely unsafe, instead of it being completely safe.
I feel the current name misrepresents the function's purpose. May I suggest renaming the function then?
It's not particularly elegant, but maybe something along the lines of `to_s_base` or `to_s_impl`?